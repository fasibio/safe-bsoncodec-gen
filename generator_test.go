package safebsoncodecgen_test

import (
	"os"
	"testing"

	safebsoncodecgen "github.com/fasibio/safe-bsoncodec-gen"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestGenerator(t *testing.T) {

	gen := safebsoncodecgen.NewBsonGenerator(&safebsoncodecgen.Config{
		PackageName: "test_model",
		PackagePath: "github.com/fasibio/safe-bsoncodec-gen/test_model",
		Direction:   "./test_model",
		FileName:    "test_gen.go",
	}, "github.com/fasibio/safe-bsoncodec-gen/test_model")
	require.NoError(t, gen.Run())
	f, err := os.ReadFile("./test_model/test_gen.go")
	require.NoError(t, err)
	snaps.MatchSnapshot(t, string(f))
	os.Remove("./test_model/test_gen.go")
}
