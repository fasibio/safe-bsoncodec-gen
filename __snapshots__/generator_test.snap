
[TestGenerator - 1]
package test_model

import (
    "reflect"

    "github.com/fasibio/safe"
    safebsoncodecgen "github.com/fasibio/safe-bsoncodec-gen"
    "go.mongodb.org/mongo-driver/bson/bsoncodec"
)

func RegisterOptionCodec(registry *bsoncodec.Registry) {
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[[10]string]{}),
        safebsoncodecgen.OptionCodec[[10]string]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[[10]string]{}),
        safebsoncodecgen.OptionCodec[[10]string]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[[]*Model2]{}),
        safebsoncodecgen.OptionCodec[[]*Model2]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[[]*Model2]{}),
        safebsoncodecgen.OptionCodec[[]*Model2]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[[]string]{}),
        safebsoncodecgen.OptionCodec[[]string]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[[]string]{}),
        safebsoncodecgen.OptionCodec[[]string]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[float64]{}),
        safebsoncodecgen.OptionCodec[float64]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[float64]{}),
        safebsoncodecgen.OptionCodec[float64]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[Model2]{}),
        safebsoncodecgen.OptionCodec[Model2]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[Model2]{}),
        safebsoncodecgen.OptionCodec[Model2]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[int]{}),
        safebsoncodecgen.OptionCodec[int]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[int]{}),
        safebsoncodecgen.OptionCodec[int]{},
    )
    registry.RegisterTypeDecoder(
        reflect.TypeOf(safe.Option[string]{}),
        safebsoncodecgen.OptionCodec[string]{},
    )
    registry.RegisterTypeEncoder(
        reflect.TypeOf(safe.Option[string]{}),
        safebsoncodecgen.OptionCodec[string]{},
    )
}

---
